pkg/morph/client/reload.go

```diff
@@ -0,0 +1,31 @@
+package client
+
+import "slices"
+
+// Reload allows runtime reconfiguration for WithEndpoints parameter.
+func (c *Client) Reload(opts ...Option) {
+	cfg := new(cfg)
+	for _, o := range opts {
+		o(cfg)
+	}
+
+	c.endpointsLock.Lock()
+
+	c.cfg.endpoints = cfg.endpoints
```

Comment:
hm, maybe not this PR's zone but why we need two field updates?

Reply:
Yeah, I wanted to fix this immediately as well. But would be nice if @End-rey untangles it.